src/modelmanager.cpp

```diff
@@ -485,46 +503,47 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
 
     auto model = getModelIfExistCreateElse(config.getName());
     getVersionsToChange(config, model->getModelVersions(), requestedVersions, versionsToStart, versionsToReload, versionsToRetire);
-    downloadModels(fs, config, versionsToStart);
-    downloadModels(fs, config, versionsToReload);
 
-    status = model->addVersions(versionsToStart, config);
-    if (!status.ok()) {
-        spdlog::error("Error occurred while loading model: {} versions; error: {}",
-            config.getName(),
-            status.string());
-        return status;
+    try {
+        downloadModels(fs, config, versionsToStart);
+
+        blocking_status = model->addVersions(versionsToStart, config);
+        if (!blocking_status.ok()) {
+            spdlog::error("Error occurred while loading model: {} versions; error: {}",
+                config.getName(),
+                blocking_status.string());
+        }
+    } catch (std::exception& e) {
+        spdlog::error("Exception occurred while loading model: {};", e.what());
```

Comment:
This function is now very long. Please encapsulate::

eg
ModelManager::addModelVersions
ModelManager::reloadModelVersions